packages/code-du-travail-api/src/server/routes/search/utils.js

```diff
@@ -16,34 +16,29 @@ const merge = (res1, res2, max_result) => {
 };
 
 // Remove Duplicates
+const predicateFn = ({ _source: { source, slug } }) => `${source}/${slug}`;
 
-const removeDuplicate = arr =>
-  Object.values(
-    arr.reduce(
-      (values, current) => ({
-        ...values,
-        [current._source.source + "/" + current._source.slug]: current
-      }),
-      {}
-    )
-  );
+const removeDuplicate = (arr, predicate = predicateFn) =>
```

Comment:
On a vraiment besoin d'un `flatMap` ?
Juste un filter avec un some dedans ça me parait un peu plus explicite pour filtrer si on n'a pas de soucis de sous tableaux !
```
arr.filter(
  (item, index) =>
    !arr.some(
      (subItem, subIndex) =>
        subIndex < index && predicate(item) === predicate(subItem)
    )
);
```

Reply:
filter is ok but I found the `some` syntax very hard to read

Reply:
It's basically the same than `.find` imho